src/UriFactory.php

```diff
@@ -16,4 +36,217 @@ public function createUri(string $uri = ''): UriInterface
     {
         return new Uri($uri);
     }
+
+    /**
+     * Create a Uri instance based on the headers and $_SERVER data.
+     *
+     * @param array<string, string|list<string>> $headers
+     * @param array $server SAPI parameters
+     */
+    public static function createFromSapi(array $server, array $headers): Uri
+    {
+        $uri = new Uri('');
+
+        // URI scheme
```

Comment:
These comments are kinda redundant with the code :D 

Perhaps better variable names could replace them?